minio/minioadminhttp.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# MinIO Python Library for Amazon S3 Compatible Cloud Storage,
+# (C) 2021 MinIO, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=too-many-public-methods
+
+"""MinIO Admin wrapper using HTTP API."""
+
+from __future__ import absolute_import
+import json
+from typing import Dict
+from minio.api import HttpClient
+from minio.crypto import decrypt, encrypt
+from minio.datatypes import UserInfo, parse_list_users
+from minio.error import AdminResponseError
+
+
+_ADMIN_PATH_PREFIX = "/minio/admin/v3"
+
+
+class MinioAdminHttp(HttpClient):
```

Comment:
Move code in this class to `MinioAdmin` class and remove `mc` usage in the class.

Reply:
Should I change the interface of the MinioAdmin class and make a breaking change?
Also, there is a lot of user commands and I wanted to implement them step by step. Is it okay to make only the subset of them available at the moment?

Reply:
Yes. There is no need to have a separate class. We have to leverage `MinioAdmin` class. APIs in `MinioAdmin` class return `dict`. You could keep the APIs returning `dict`.

We would need to have current APIs available by http client. May be you could send additional changes to this PR as new commits.

Reply:
Ok, I get it. I will change this PR and remove changes to api.py.

Do I need to implement all methods in the current API or can I start with only the basic ones?

Reply:
We need to have all methods, but you could implement one after another in this PR as a new commits